datalake/src/main/java/com/sequenceiq/datalake/flow/dr/validation/DatalakeBackupValidationActions.java

```diff
@@ -0,0 +1,174 @@
+import org.springframework.context.annotation.Configuration;
+import org.springframework.statemachine.action.Action;
+
+import com.dyngr.exception.PollerStoppedException;
+import com.sequenceiq.cloudbreak.common.exception.CloudbreakServiceException;
+import com.sequenceiq.cloudbreak.datalakedr.model.DatalakeBackupStatusResponse;
+import com.sequenceiq.cloudbreak.event.ResourceEvent;
+import com.sequenceiq.datalake.entity.DatalakeStatusEnum;
+import com.sequenceiq.datalake.entity.SdxCluster;
+import com.sequenceiq.datalake.flow.SdxContext;
+import com.sequenceiq.datalake.flow.SdxEvent;
+import com.sequenceiq.datalake.flow.dr.backup.DatalakeBackupFailureReason;
+import com.sequenceiq.datalake.flow.dr.validation.event.DatalakeBackupValidationFailedEvent;
+import com.sequenceiq.datalake.flow.dr.validation.event.DatalakeBackupValidationWaitRequest;
+import com.sequenceiq.datalake.flow.dr.validation.event.DatalakeTriggerBackupValidationEvent;
+import com.sequenceiq.datalake.metric.MetricType;
+import com.sequenceiq.datalake.metric.SdxMetricService;
+import com.sequenceiq.datalake.service.AbstractSdxAction;
+import com.sequenceiq.datalake.service.sdx.SdxService;
+import com.sequenceiq.datalake.service.sdx.dr.SdxBackupRestoreService;
+import com.sequenceiq.datalake.service.sdx.status.SdxStatusService;
+import com.sequenceiq.flow.core.Flow;
+
+@Configuration
+public class DatalakeBackupValidationActions {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(DatalakeBackupValidationActions.class);
+
+    private static final String BACKUP_ID = "BACKUP-ID";
+
+    private static final String REASON = "REASON";
+
+    @Inject
+    private SdxBackupRestoreService sdxBackupRestoreService;
+
+    @Inject
+    private SdxStatusService sdxStatusService;
+
+    @Inject
+    private SdxMetricService metricService;
+
+    @Inject
+    private SdxService sdxService;
+
+    @Bean(name = "DATALAKE_TRIGGERING_BACKUP_VALIDATION_STATE")
+    public Action<?, ?> triggerDatalakeBackupValidationAction() {
+        return new AbstractSdxAction<>(DatalakeTriggerBackupValidationEvent.class) {
+            @Override
+            protected void doExecute(SdxContext context, DatalakeTriggerBackupValidationEvent payload, Map<Object, Object> variables) {
+                LOGGER.info("Triggering datalake backup validation for {}", payload.getResourceId());
+                variables.put(REASON, payload.getReason().name());
+
+                SdxCluster sdxCluster = sdxService.getById(payload.getResourceId());
+                sdxStatusService.setStatusForDatalakeAndNotify(DatalakeStatusEnum.RUNNING,
+                        ResourceEvent.DATALAKE_BACKUP_VALIDATION_IN_PROGRESS,
+                        "Datalake backup validation in progress", payload.getResourceId());
+                metricService.incrementMetricCounter(MetricType.SDX_BACKUP_VALIDATION_REQUESTED, sdxCluster);
+
+                DatalakeBackupStatusResponse backupStatusResponse =
```

Comment:
If its a long running task, e.g: a http/grcp call, it should be moved to a Handler class.
See: https://github.com/hortonworks/cloudbreak/tree/master/flow#handlers

Reply:
Should it? There are plenty of other places that do this exact same thing. Also all this does is fire a GRPC call without waiting for it, instead we opt to have a waiting process as a separate step in the flow.
Hopefully that makes sense.